File: Data Structures & Algorithms/missing-number/submission-0.cpp
class Solution {
public:
    int missingNumber(vector<int>& nums) {
        std::unordered_set<int> s;
        for (int i: nums) {
            s.insert(i);
        }

        for (int i = 0; i <= nums.size(); i++) {
            if (!s.contains(i)) {
                return i;
            }
        }

        return -1;
    }
};

// 0000 0001
// 

File: Data Structures & Algorithms/find-minimum-in-rotated-sorted-array/submission-0.py
class Solution:
    def findMin(self, nums: List[int]) -> int:
        l = 0
        r = len(nums) - 1

        while l < r:
            m = (l + r) // 2

            if nums[m] > nums[r]:
                l = m + 1
            if nums[m] < nums[r]:
                r = m

        return nums[l]                
        

# 3, 4, 5, 1, 2
# l     m     r

# 1, 2
# l  r
# m 